### DISCUSSION TITLE
DegreesToRadians rounding error?

### DISCUSSION THREAD

**Discussion by lin-ycv** [visose/Robots#146] [Q&A] [answered]
I'm wondering why the rounding is required here?
https://github.com/visose/Robots/blob/3d59a0213ab6069bb15227de990e50e196d8769b/src/Robots.Grasshopper/RobotSystem/DegreesToRadians.cs#L33
we could simply do:
`string radiansText = string.Join(",", radians);`

tho it doesn't really matter 99% of the time, for the times when we want to control a robot using FK instead of IK, this rounding can cause issues when we want to move near the min/max range of the axis

**Comment by visose** [comment, marked as answer]
I guess the the use for text representing radians should be limited to:
- A human writing the values manually.
- A human reading the values.
- Generating a robot program text file with hardcoded values.

Right now, using Grasshopper components, you can only create a joint target using a comma separated string. This assumes the user is writing the values manually. I was thinking of replacing the string input with something like a "joints" parameter that uses doubles directly. The casting of the comma separated string will create this joint parameter. The degree to radians component can output directly a joints parameter. There are some things to consider, like if it should be an array of doubles, or a Vector6 struct of doubles, but then what about external axis? Some robots use two vector6 structures, one for robot axis and one for external axis. I'll think about this.

For now, you can avoid the string by using a scripting component. You can do the degree to joint conversion and use the result to create a joint target without using strings.

**Reply by visose** [reply]
In the latest update there's a new Joints parameter as described above (internally its a double array for now).
